pre_commit_hooks/linting.sh

```diff
@@ -0,0 +1,152 @@
+RUN_PYLINT=TRUE
+
+####################################
+# Overwrite linter versions if set #
+####################################
+
+if [ $(yq '.jobs.lint.with | has("black-version")' $CODE_LINTING_WORKFLOW) == true ]
+then
+    DEFAULT_BLACK_VERSION=$(cat /requirements.txt | grep black== | cut -d "=" -f 3)
+    NEW_BLACK_VERSION=$(yq '.jobs.lint.with."black-version"' $CODE_LINTING_WORKFLOW)
+    if [ "$(echo $NEW_BLACK_VERSION | tr '"' 'x')" = "xx" ]
+    then
+        RUN_BLACK=FALSE
+        echo "- black configured to be skipped (empty string)"
+    elif [ $DEFAULT_BLACK_VERSION != $NEW_BLACK_VERSION ]
+    then
+        # sed would fail with Permission denied
+        # sed -i "s+$DEFAULT_BLACK_VERSION+$NEW_BLACK_VERSION+g" /requirements.txt
+        echo "- Warning: overwritten black version will not be installed!"
+    fi
+else
+    echo "- black version not overwritten in code workflow"
+fi
+
+if [ $(yq '.jobs.lint.with | has("flake8-version")' $CODE_LINTING_WORKFLOW) == true ]
+then
+    DEFAULT_FLAKE8_VERSION=$(cat /requirements.txt | grep flake8== | cut -d "=" -f 3)
+    NEW_FLAKE8_VERSION=$(yq '.jobs.lint.with."flake8-version"' $CODE_LINTING_WORKFLOW)
+    if [ "$(echo $NEW_FLAKE8_VERSION | tr '"' 'x')" = "xx" ]
+    then
+        RUN_FLAKE8=FALSE
+        echo "- flake8 configured to be skipped (empty string)"
+    elif [ $DEFAULT_FLAKE8_VERSION != $NEW_FLAKE8_VERSION ]
+    then
+        # sed would fail with Permission denied
+        # sed -i "s+$DEFAULT_FLAKE8_VERSION+$NEW_FLAKE8_VERSION+g" /requirements.txt
+        echo "- Warning: overwritten flake8 version will not be installed!"
+    fi
+else
+    echo "- flake8 version not overwritten in code workflow"
+fi
+
+if [ $(yq '.jobs.lint.with | has("pylint-version")' $CODE_LINTING_WORKFLOW) == true ]
+then
+    DEFAULT_PYLINT_VERSION=$(cat /requirements.txt | grep pylint== | cut -d "=" -f 3)
+    NEW_PYLINT_VERSION=$(yq '.jobs.lint.with."pylint-version"' $CODE_LINTING_WORKFLOW)
+    if [ "$(echo $NEW_PYLINT_VERSION | tr '"' 'x')" = "xx" ]
+    then
+        RUN_PYLINT=FALSE
+        echo "- pylint configured to be skipped (empty string)"
+    elif [ $DEFAULT_PYLINT_VERSION != $NEW_PYLINT_VERSION ]
+    then
+        # sed would fail with Permission denied
+        # sed -i "s+$DEFAULT_PYLINT_VERSION+$NEW_PYLINT_VERSION+g" /requirements.txt
+        echo "- Warning: overwritten pylint version will not be installed!"
+    fi
+else
+    echo "- pylint version not overwritten in code workflow"
+fi
+
+echo
+
+###################
+# Install linters #
+###################
+# TODO: if versions are not default versions, this fails due to missing permissions
+pip3 install -r /requirements.txt
+echo
+echo "flake8: `flake8 --version`"
+echo "pylint: `pylint --version`"
+echo "black: `black --version`"
+
+########
+# lint #
+########
+
+cd /src
+
+if [ $RUN_FLAKE8 != "FALSE" ]
+then
+    echo
+    echo "FLAKE8:"
+    flake8 --config=/src/.flake8 --count --statistics --show-source .
```

Comment:
Why is `.flake8` in the `/src` folder and not in `./.flake8` ? Or is `.` = `/src` ? And why is not `CODE_REPO_PATH` used?

Reply:
Good catch - updated it to `cd` with variable usage and then relative paths after that.